src/VirtualClient/VirtualClient.Dependencies/Packaging/GitRepoClone.cs

```diff
@@ -100,6 +117,25 @@ await systemManagement.PackageManager.RegisterPackageAsync(
                 await systemManagement.PackageManager.RegisterPackageAsync(package, cancellationToken)
                     .ConfigureAwait(false);
             }
+
+            if (this.Checkout != string.Empty)
+            {
+                using (IProcessProxy checkoutProcess = processManager.CreateProcess("git", $"-C {cloneDirectory} checkout {this.Checkout}", this.PlatformSpecifics.PackagesDirectory))
```

Comment:
Use ExecuteCommandAsync on VirtualClientCompnentExtensions unless you are doing something special